internal/goroutine/recorder/recorder.go

```diff
@@ -0,0 +1,234 @@
+	return &Recorder{rcache: cache, logger: logger, hostName: hostName}
+}
+
+// Register registers a new routine with the recorder.
+func (m *Recorder) Register(r Recordable) {
+	m.recordables = append(m.recordables, r)
+}
+
+// RegistrationDone should be called after all recordables have been registered.
+// It saves the known job names, host names, and routine names in Redis, along with updating their “last seen” date/time.
+func (m *Recorder) RegistrationDone() {
+	// Save/update known job names
+	for _, jobName := range m.collectAllJobNames() {
+		m.saveKnownJobName(jobName)
+	}
+
+	// Save known host name
+	m.saveKnownHostName()
+
+	// Save/update all known recordables
+	for _, r := range m.recordables {
+		m.saveKnownRoutine(r)
+	}
+}
+
+// collectAllJobNames collects all known job names in Redis.
+func (m *Recorder) collectAllJobNames() []string {
+	var allJobNames []string
+	for _, r := range m.recordables {
+		jobName := r.JobName()
+		if slices.Contains(allJobNames, jobName) {
+			continue
+		}
+		allJobNames = append(allJobNames, jobName)
+	}
+
+	return allJobNames
+}
+
+// saveKnownJobName updates the “lastSeen” date of a known job in Redis. Also adds it to the list of known jobs if it doesn’t exist.
+func (m *Recorder) saveKnownJobName(jobName string) {
+	err := m.rcache.SetHashItem("knownJobNames", jobName, time.Now().Format(time.RFC3339))
+	if err != nil {
+		m.logger.Error("failed to save/update known job name", log.Error(err), log.String("jobName", jobName))
+	}
+}
+
+// saveKnownHostName updates the “lastSeen” date of a known host in Redis. Also adds it to the list of known hosts if it doesn’t exist.
+func (m *Recorder) saveKnownHostName() {
+	err := m.rcache.SetHashItem("knownHostNames", m.hostName, time.Now().Format(time.RFC3339))
+	if err != nil {
+		m.logger.Error("failed to save/update known host name", log.Error(err), log.String("hostName", m.hostName))
+	}
+}
+
+// saveKnownRoutine updates the routine in Redis. Also adds it to the list of known recordables if it doesn’t exist.
+func (m *Recorder) saveKnownRoutine(recordable Recordable) {
+	r := serializableRoutineInfo{
+		Name:        recordable.Name(),
+		Type:        recordable.Type(),
+		JobName:     recordable.JobName(),
+		Description: recordable.Description(),
+		Interval:    recordable.Interval(),
+	}
+
+	// Serialize Routine
+	routineJson, err := json.Marshal(r)
+	if err != nil {
+		m.logger.Error("failed to serialize routine", log.Error(err), log.String("routineName", r.Name))
+		return
+	}
+
+	// Save/update Routine
+	err = m.rcache.SetHashItem("knownRoutines", r.Name, string(routineJson))
+	if err != nil {
+		m.logger.Error("failed to save/update known routine", log.Error(err), log.String("routineName", r.Name))
+	}
+}
+
+// LogStart logs the start of a routine.
+func (m *Recorder) LogStart(r Recordable) {
+	m.rcache.Set(r.Name()+":"+m.hostName+":"+"lastStart", []byte(time.Now().Format(time.RFC3339)))
+	m.logger.Debug("" + r.Name() + " just started! 🚀")
```

Comment:
I'd change the log messages in here to make use of structured logging:

```suggestion
	m.logger.Debug("Routine just started! 🚀", log.String("routine", r.Name()))
```